### ISSUE TITLE
Add "nuc" branch label option parallel to "aa" 

### ISSUE THREAD

**Issue by corneliusroemer** [nextstrain/auspice#1736]
### Context
Automatic "aa" branch labels are useful. Sometimes it would be nice to show "nuc" branch labels instead, when there are branches without "aa"s in particular.

### Description
Add "nuc" branch labels the way we have "aa" branch labels at the moment.

<img width="501" alt="image" src="https://github.com/nextstrain/auspice/assets/25161793/22c9bc70-5f61-490c-8bd4-82b4db0a74e5">

**Comment by jameshadfield** [comment]
> Add "nuc" branch labels the way we have "aa" branch labels at the moment.

The way we do it now is for branch labels to be defined in the JSON, i.e. they're separate from the mutations defined on the branch. This isn't very efficient, but to see what this looks like for your dataset you can (and perhaps should) add these labels via a python script.

I'd be in favor of allowing Auspice to automatically create aa & nuc labels from the mutations, thus forgoing the need for explicit branch labels for mutations.

**Comment by corneliusroemer** [comment]
I thought the `aa` are done automatically in a way that `nuc`s aren't? Maybe the automatic part for `aa` is in some augur command as opposed to auspice, is that what you're saying? There's a difference between manual branch labels like clade and whatnot and the magic "aa" which is always smaller font.

**Comment by jameshadfield** [comment]
> Maybe the automatic part for aa is in some augur command as opposed to auspice, is that what you're saying?

Yes. [Here's](https://github.com/nextstrain/augur/blob/89c6b51bbc0d64080908113e1be4baad3fd433e0/augur/export_v2.py#L1020-L1029) the relevant code within `augur export v2`

> the magic "aa" which is always smaller font.

Oh, yes you're right:

https://github.com/nextstrain/auspice/blob/40befc6ea4ac02226db2f5658faa12beda52a3e1/src/components/tree/phyloTree/labels.js#L57-L60

**Comment by corneliusroemer** [comment]
So it's a combination of special treatment by augur _and_ auspice.

Any reason we should auto-export nucs in augur export? Though I should probably transfer this issue over to augur then...

**Comment by jameshadfield** [comment]
> Any reason we should auto-export nucs in augur export? Though I should probably transfer this issue over to augur then...

I'd prefer to (or someone to) update Auspice to auto-generate these branches without needing to further bloat the JSON with duplicate nuc + aa mutation data. It shouldn't be too hard. In the meantime however adding a "nuc" branch label via a small script will allow you to see these data on the tree. Does that sound reasonable? Happy for others to weigh in here.

**Comment by joverlee521** [comment]
This feature was brought up again in [Slack](https://bedfordlab.slack.com/archives/C01LCTT7JNN/p1779465679693839) in the context of recent Hanta and BDBV outbreaks.

**Comment by jameshadfield** [comment]
There's a few ideas that have been kicking around lately (including this one) which I think can be solved in a single push, or at least considered together when designing things if we want to tackle them separately.

1. Show masked regions in entropy panel. We often mask regions for tree building, so find a way to export these from Augur and show then in the entropy panel.
2. Allow masks to be toggled on/off, and have the entropy panel bars and the tree tooltip/info boxes hide mutations in masked regions.
3. Move mutation branch labels into Auspice (i.e. out of the JSON), and also have them respect the mask toggle state

Additionally, but a little more removed:

4. When hovering on branches / tips, explore having a mini-genome view (skeleton-like) with vertical bars showing mutations. This would help flag problematic regions, ADAR/APOBEC edits, recombination/reassortment.
5. Show segments in the entropy panel. You could imaging re-numbering mutations relative to a segment, but this seems overkill; or clicking on a segment à la a CDS, but this also seems overkill.
